### PR TITLE
Update the Profile display page to render URLs as a-tags

### PR DESCRIPTION
**Pull request by Xiaoming**

### CODE REVIEW

File: spec/features/person_profile_spec.rb [hunk 1]
```diff
@@ -75,14 +76,36 @@
       click_link "My Profile"
       page.should have_css("img[class$='gravatar']")
     end
+
+    it 'should render personal webpage and blog URLs as a-tags' do
```
Comment (jcoyne) [inline spec/features/person_profile_spec.rb]: This should probably be a view test, not a feature test.

File: app/views/curate/people/show.html.erb [hunk 2]
```diff
@@ -21,7 +21,13 @@
       <% if @person.send(attribute_name).present? %>
         <dt><%="#{derived_label_for( @person, attribute_name) }:" %></dt>
         <% [@person.send(attribute_name)].flatten.compact.each do |value| %>
-        <dd><%= value %></dd>
```
Comment (jcoyne) [inline app/views/curate/people/show.html.erb:24]: This is probably too much code to have in a view. Can you put it in a helper method instead?

**Comment by jcoyne** [comment]
This seems like a great place to target a javascript injection attack.  Can you write a test showing that it's not exploitable?

File: spec/features/person_profile_spec.rb (hunk 1 above)
Comment (Xiaoming) [inline spec/features/person_profile_spec.rb]: Thanks, Justin. How can we tell if a test should be a view test or a feature test? The test I wrote needs to update user's account to show the url link, so I thought it's more like other feature tests in person_profile_spec.rb file, such as 'show gravatar image if profile image not uploaded'.

On Jan 4, 2014, at 12:01 AM, Justin Coyne wrote:

In spec/features/person_profile_spec.rb:

> @@ -75,14 +76,36 @@
>        click_link "My Profile"
>        page.should have_css("img[class$='gravatar']")
>      end
> +
> -    it 'should render personal webpage and blog URLs as a-tags' do

This should probably be a view test, not a feature test.

Comment (jcoyne) [inline spec/features/person_profile_spec.rb]: View tests are many times faster than spec tests.  The current curate builds have been bumping up against the time limit in Travis, so whenever you can test something with a view test, you should.  A feature test does not necessarily need to test all the edge cases, it should be more general.  For example: "I can update my profile"

Comment (Xiaoming) [inline spec/features/person_profile_spec.rb]: ok. thanks

On Jan 6, 2014, at 5:05 PM, Justin Coyne wrote:

In spec/features/person_profile_spec.rb:

> @@ -75,14 +76,36 @@
>        click_link "My Profile"
>        page.should have_css("img[class$='gravatar']")
>      end
> +
> -    it 'should render personal webpage and blog URLs as a-tags' do

View tests are many times faster than spec tests. The current curate builds have been bumping up against the time limit in Travis, so whenever you can test something with a view test, you should. A feature test does not necessarily need to test all the edge cases, it should be more general. For example: "I can update my profile"

File: app/views/curate/people/show.html.erb (hunk 2 above)
Comment (jeremyf) [inline app/views/curate/people/show.html.erb:24]: +1 for helper method.

**Comment by jeremyf** [comment]
@Xiaoming is going to be pushing up a Regexp spec as a matter of practice. We are going to review it then proceed with rails_autolink

**Comment by jeremyf** [comment]
The specs look to be testing the right thing and I believe the exercise is done. With #409 this pull request won't be merged.
